syntax = "proto3";

import "google/protobuf/empty.proto";

package bookstore;

message Book {
  string id = 1;
  string title = 2;
  string author = 3;
  string description = 4;
  float price = 5;
}

service BookService {
  rpc CreateBook(Book) returns (Book);
  rpc ReadBook(ReadBookRequest) returns (Book);
  rpc UpdateBook(Book) returns (Book);
  rpc DeleteBook(DeleteBookRequest) returns (google.protobuf.Empty);
}

message ReadBookRequest {
  string id = 1;
}

message DeleteBookRequest {
  string id = 1;
}
